agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -3854,7 +3853,7 @@ bool slave_thread::handle_autoconfiguration_wsc(Socket *sd, ieee1905_1::CmduMess
     LOG(DEBUG) << "Received AP_AUTOCONFIGURATION_WSC_MESSAGE";
 
     std::shared_ptr<wfa_map::tlvApRadioIdentifier> ruid = nullptr;
-    std::vector<std::shared_ptr<ieee1905_1::tlvWscM2>> m2_list;
+    std::vector<std::shared_ptr<WSC::m2>> m2_list;
```

Comment:
As a general rule of thumb, unless there's a need for index based access to elements, it's better to use `std::list` as it's more predictable in runtime complexity and memory allocations.

Reply:
Will change it in a separate commit.